File: practica4/AreaFiguras.py
import math
class AreasFiguras:
    # Círculo
    def calcular_area_circulo(self, radio):
        return math.pi * radio ** 2

    # Rectángulo
    def calcular_area_rectangulo(self, base, altura):
        return base * altura

    # Triángulo rectángulo
    def calcular_area_triangulo(self, base, altura, es_triangulo):
        if es_triangulo:
            return (base * altura) / 2
        else:
            return 0

    # Trapecio
    def calcular_area_trapecio(self, base_mayor, base_menor, altura):
        return ((base_mayor + base_menor) * altura) / 2

    # Pentágono
    def calcular_area_pentagono(self, lado, apotema):
        perimetro = lado * 5
        return (perimetro * apotema) / 2

figuras = AreasFiguras()

radio = 5

b_rect = 4
alt_rect = 6

b_tri = 3
alt_tri = 4

b_mayor_trap = 6
b_menor_trap = 4
alt_trap = 5

l_pent = 4
apotema_pent = 3

area_circulo = figuras.calcular_area_circulo(radio)
area_rectangulo = figuras.calcular_area_rectangulo(b_rect, alt_rect)
area_triangulo = figuras.calcular_area_triangulo(b_tri, alt_tri, True)
area_trapecio = figuras.calcular_area_trapecio(b_mayor_trap, b_menor_trap, alt_trap)
area_pentagono = figuras.calcular_area_pentagono(l_pent, apotema_pent)

print(f"Área del círculo: {area_circulo}")
print(f"Área del rectángulo: {area_rectangulo}")
print(f"Área del triángulo rectángulo: {area_triangulo}")
print(f"Área del trapecio: {area_trapecio}")
print(f"Área del pentágono: {area_pentagono}")

File: practica4/AreaFiguras.java
package nead1;
public class AreasFiguras {

    //círculo
    public double calcularArea(double radio) {
        return Math.PI * Math.pow(radio, 2);  
    }

    //rectángulo
    public double calcularArea(double base, double altura) {
        return base * altura;
    }

    //triángulo rectángulo
    public double calcularArea(double base, double altura, boolean esTriangulo) {
        if (esTriangulo) {
            return (base * altura) / 2;  
        } else {
            return 0;
        }
    }
    //trapecio
    public double calcularArea(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2;  
    }

    //pentágono
    public double calcularArea(int lado, double apotema) {
        double perimetro = lado * 5; 
        return (perimetro * apotema) / 2;  
    }

    public static void main(String[] args) {
        AreasFiguras figuras = new AreasFiguras();

        
        double radio = 5; 
        
        double bRect = 4;  
        double altRect = 6;
        
        double bTri = 3; 
        double altTri = 4;
       
        double bMayorTrap = 6;  
        double bMenorTrap = 4;  
        double altTrap = 5;
        
        int lPent = 4;  
        double apotemaPent = 3;

        
        double areaCirculo = figuras.calcularArea(radio);
        double areaRectangulo = figuras.calcularArea(bRect, altRect);
        double areaTriangulo = figuras.calcularArea(bTri, altTri, true);
        double areaTrapecio = figuras.calcularArea(bMayorTrap, bMenorTrap, altTrap);
        double areaPentagono = figuras.calcularArea(lPent, apotemaPent);

        System.out.println("Area del circulo: " + areaCirculo);
        System.out.println("Area del rectangulo: " + areaRectangulo);
        System.out.println("Area del triangulo rectangulo: " + areaTriangulo);
        System.out.println("Area del trapecio: " + areaTrapecio);
        System.out.println("Area del pentagono: " + areaPentagono);
    }
}
